scripts/pi-hole/js/index.js

```diff
@@ -313,7 +313,7 @@ function updateClientsOverTime() {
         clientsChart.data.datasets[0].pointHoverRadius = 5;
         clientsChart.data.datasets[0].label = labels[0];
 
-        for (i = clientsChart.data.datasets.length; i < plotdata[0].length; i++)
+        for (i = clientsChart.data.datasets.length; plotdata.length && i < plotdata[0].length; i++)
```

Comment:
How does your change actually prevent the error you are seeing from happening? When looking at
```javascript
plotdata.length && i < plotdata[0].length
```
I'd think that `plotdata[0].length` is accessed whatever `plotdata.length` is to check the condition...

Reply:
`plotdata.length` is coerced to true or false based on its value. `0` is false and anything else is true. So if the length is 0 it doesn't run the loop. It can be changed to `plotdata.length === 0` if you'd like.

Reply:
That's not what I was after. Your modifications imply the following code:
```javascript
plotdata.length && i < plotdata[0].length
```
is checked every time the loop executes. However, this will (for an empty data set) turn out to be:
```
0 && undefined          -> false
```
However, the `error` will still be there. If you want to kill off this error, the code modification should probably look more like:
```javascript
if(plotdata.length > 0) // whereas the "> 0" can be omitted as you already suggested
{
    for (i = clientsChart.data.datasets.length; i < plotdata[0].length; i++)
    {
        [...]
    }
}
```

Reply:
The boolean `&&` shortcuts the right side if the left side is false, so if `plotdata.length` is 0 it will not evaluate the right side, and will instead just return false for the whole thing.